packages/api-plugin-fulfillment/migrations/3.js

```diff
@@ -0,0 +1,56 @@
+const FF_TYPE = "shipping";
+const FF_METHOD = "flatRate";
+const COLL_DEST = "Fulfillment";
+
+/**
+ * @summary Performs migration up from previous data version
+ * @param {Object} context Migration context
+ * @param {Object} context.db MongoDB `Db` instance
+ * @param {Function} context.progress A function to report progress, takes percent
+ *   number as argument.
+ * @return {undefined}
+ */
+async function up({ db, progress }) {
+  progress(0);
+
+  const operations = [];
+
+  const ffTypeUpdate = {
+    updateMany: {
+      filter: { fulfillmentType: { $exists: false } },
+      update: {
+        $set: {
+          fulfillmentType: FF_TYPE
+        }
+      }
+    }
+  };
+
+  const ffMethodUpdate = {
+    updateMany: {
+      filter: { methods: { $exists: true } },
+      update: {
+        $set: {
+          "methods.$[eachMethod].fulfillmentMethod": FF_METHOD
+        }
+      },
+      arrayFilters: [
+        {
+          "eachMethod.fulfillmentMethod": { $exists: false }
+        }
+      ]
+    }
+  };
+
+  operations.push(ffTypeUpdate);
+  operations.push(ffMethodUpdate);
+
+  const bulkWriteResp = await db.collection(COLL_DEST).bulkWrite(operations);
+  if (bulkWriteResp.writeErrors && bulkWriteResp.writeErrors.length) throw new Error("Error while updating Fulfillment collection");
+  progress(100);
+}
+
+export default {
+  down: "impossible", // We are not tracking the updated documents, hence cannot revert
```

Comment:
It seems like migrating down might not really be necessary (old code would work either way). If so, you could use `"unnecessary"` rather than `"impossible"` here, which would then not block rollback.

Reply:
Thanks for the suggestion, that approach is correct. I am making this change

Reply:
Done